pkg/registry/apis/query/register.go

```diff
@@ -170,6 +178,14 @@ func (b *QueryAPIBuilder) UpdateAPIGroupInfo(apiGroupInfo *genericapiserver.APIG
 
 	storage := map[string]rest.Storage{}
 
+	// Get a list of all datasource instances
+	if b.features.IsEnabledGlobally(featuremgmt.FlagGrafanaAPIServerWithExperimentalAPIs) {
```

Comment:
I think we'd gain from having an explicit flag for this. as we start deploying this to cloud environments, I don't want to accidentally enable any other experimental features behind the generic experimental flag.

maybe `FlagGrafanaAPIServerWithDSConfigAPIs`? or, a better name along those lines?